components/NVS/src/Storage.cpp

```diff
@@ -100,6 +100,18 @@ esp_err_t NVS::get_bool(const char* key, bool& value)
     return ESP_OK;
 }
 
+esp_err_t NVS::setString(std::string key, std::string data) 
+{
+    return nvs_set_str(m_handle, key.c_str(), data.c_str());
+}
```

Comment:
```suggestion
esp_err_t NVS::setString(const char* key, std::string data) 
{
    return nvs_set_str(m_handle, key, data.c_str());
}
```

Reply:
`std::string` allocates memory in the **heap**, here we don't need that for the key.